### PR TITLE
Assets Confirmation Flow

### PR DESCRIPTION
**Pull request by pivilartisant**
Complete asset import flow. 

Major changes : input form is now part of AssetsModalImport -> I change the level because I needed to send info to the ImportResult screen... it was so we can have less params

Screenshot 2023-08-16 at 14 18 28

### CODE REVIEW

File: web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx [hunk 1]
```diff
@@ -1,32 +1,169 @@
...
+      default:
+        setInputError({
+          address: Intl.t('assets.import.failure-screen.unkown-error'),
+        });
+    }
+  }
+  function isValidAssetAddress(input: string): boolean {
+    const regexPattern = /^AS[0-9a-zA-Z]+$/;
+    return regexPattern.test(input);
+  }
```
Comment (Thykof) [inline web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx:84]: this function should not be here. We have a collection of validation functions here https://github.com/massalabs/station-massa-wallet/tree/6c317bd32a8608bf55ce9d772acf4092931b0e83/web-frontend/src/validation

File: web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx [hunk 2]
```diff
@@ -1,32 +1,169 @@
...
-        </div>
-      </PopupModalHeader>
-      <PopupModalContent>
-        <ImportForm {...props} />
-      </PopupModalContent>
-    </PopupModal>
+    <>
+      {importResult !== null ? (
+        <PopupModal
+          customClass="w-[513px] h-[440px]"
```
Comment (Thykof) [inline web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx:119]: why 513px?

**Review by Thykof — CHANGES_REQUESTED** [review]
I think that components are too much bound together.

File: web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx (hunk 2 above)
Comment (pivilartisant) [reply web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx:119]: This is the design size

File: web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx (hunk 1 above)
Comment (pivilartisant) [reply web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx]: moved it to the right spot !

File: web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx [hunk 3]
```diff
@@ -1,32 +1,169 @@
...
 export function AssetsImportModal({ ...props }) {
   const { closeModal } = props;
 
+  const [importResult, setImportResult] = useState<boolean | null>(null);
+  const [inputError, setInputError] = useState<InputsErrors | null>(null);
+  const [tokenAddress, setTokenAddress] = useState<string>('');
+
+  const { nickname } = useParams();
+
+  const { refetch } = useResource<IToken[]>(`accounts/${nickname}/assets`);
```
[the remark this answers is not in the document]
Comment (pivilartisant) [reply web-frontend/src/pages/Assets/AssetImportModal/AssetsImportModal.tsx]: also changed whatever had 'token' to asset

File: web-frontend/src/pages/Assets/AssetImportModal/ImportResult.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import { maskAddress } from '@/utils';
+
+interface ImportResultProps {
+  closeModal: () => void;
+  data: Asset;
+  importResult: boolean;
+  inputError: InputsErrors | null;
+}
+
+export function ImportResult(props: ImportResultProps) {
```
Comment (mazmassa) [inline web-frontend/src/pages/Assets/AssetImportModal/ImportResult.tsx:15]: I believe we have an anti patter here in terms of UI. We should sync with Lucas/Ena about it!

Comment (longgamma-massa) [reply web-frontend/src/pages/Assets/AssetImportModal/ImportResult.tsx:15]: @pivilartisant We agreed to go with toast for now. Mario will sync with you.

Comment (mazmassa) [reply web-frontend/src/pages/Assets/AssetImportModal/ImportResult.tsx:15]: Hey Paul, just synced with Ena about it. Let's keep as it is, with Toast messages (with the good information, of course) and we'll sync with Lucas to find a solution for that.

**Review by Thykof — APPROVED** [review]
LGTM

**Review by Thykof — APPROVED** [review]

**Comment by Thykof** [comment]
I merge a PR that fix some sonar lint issue (wails frontend)

**Review by mazmassa — APPROVED** [review]

**Pull request merged**